### PR TITLE
Enabling live data features in the qiskit-ibmq-provider

### PR DESCRIPTION
**Pull request by cbjuan**
### Summary
This PR adds support to LiveData in the qiskit-ibmq-provider. Specifically, it adds a new tab in the `backend` Jupyter-related widget and supports the execution of jobs with the `live_data_enabled=True` parameter in allowed backends. 

In this new tab, a user can observe which jobs have LiveData enabled and their real-time behavior while executing.

### Details and comments
A minimal example of a job submitted to an IBM Quantum backend with LiveData enabled

```
from qiskit import *
qc = QuantumCircuit(2)
qc.h(0)
qc.x(1)

job = execute(qc,
              backend=backend,
              live_data_enabled=True)
```

Overview of the `backend` widget with the new tab:
Screenshot_2022-01-18_at_13_22_37
Screenshot_2022-01-18_at_13_22_50

It works as expected in JupyterLab, Jupyter Notebook, and the IBM Quantum Notebooks platform.
Co-authored by @filemaster

### CODE REVIEW

**Comment by cbjuan** [comment]
Thanks for your review, @rathishcholarajan. We have fixed the checks and added the required release note. 

Do not hesitate to ping me if there is any other thing needed.

File: releasenotes/notes/enable-live-data-b53b1000997845da.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+features:
+  - |
+    The qiskit-ibmq-provider package now supports IBM Quantum LiveData features.
+    These features allow users to observe the real-time behavior of IBM Quantum
+    backends while executing jobs. Specifically, the provider now includes a
+    new tab in the backend Jupyter-related widget and supports the execution of
+    jobs (via `run()` or `execute()` methods) with the `live_data_enabled=True`
```
Comment (rathishcholarajan) [inline releasenotes/notes/enable-live-data-b53b1000997845da.yaml:8]: We should add the doc links for run and execute methods here. It'll look something like :meth:`qiskit.ibmq.providers.IBMQBackend.run`. You can refer to other release notes as example.

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
@cbjuan Thanks! 

The push tests on your fork are failing https://github.com/cbjuan/qiskit-ibmq-provider/runs/4964386570?check_suite_focus=true. Could you please set the following repository secret in your branch and then commit to your fork or rebuild so that we can see if all the push tests succeed. We need the push tests also to pass before merging. Push tests run more tests online integration tests than the PR checks (which are mostly unit tests).
1. Go to https://github.com/cbjuan/qiskit-ibmq-provider/settings/secrets/actions
2. Add "New repository secret"
QE_TOKEN=your prod IBM Quantum API token

**Review by rathishcholarajan — APPROVED** [review]
LGTM!

**Pull request merged**